### ISSUE TITLE
Android O: java.lang.IllegalStateException: Not allowed to start service Intent

### ISSUE THREAD

**Issue by mkeresztes** [mkeresztes/AndiCar#14]
see [Background Execution Limits](https://developer.android.com/about/versions/oreo/background.html#migration) for solutions
```
Fatal Exception: java.lang.RuntimeException: Unable to create application org.andicar2.activity.AndiCar: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.andicar2.activity/andicar.n.service.ToDoNotificationService (has extras) }: app is in background uid UidRecord{6ca9843 u0a99 RCVR idle procs:1 seq(0,0,0)}
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5794)
       at android.app.ActivityThread.-wrap1(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6541)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
Caused by java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.andicar2.activity/andicar.n.service.ToDoNotificationService (has extras) }: app is in background uid UidRecord{6ca9843 u0a99 RCVR idle procs:1 seq(0,0,0)}
       at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1505)
       at android.app.ContextImpl.startService(ContextImpl.java:1461)
       at android.content.ContextWrapper.startService(ContextWrapper.java:644)
       at andicar.n.broadcastreceiver.ServiceStarter.startServices(ServiceStarter.java:49)
       at org.andicar2.activity.AndiCar.onCreate(AndiCar.java:122)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1118)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5791)
       at android.app.ActivityThread.-wrap1(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1661)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6541)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

**Comment by mkeresztes** [comment]
fixed in 17.10-2

**Comment by mkeresztes** [comment]
fixed in 17.10-4
